ml4h/explorations.py

```diff
@@ -815,24 +867,73 @@ def infer_stats_from_segmented_regions(args):
     _, _, generate_test = test_train_valid_tensor_generators(**args.__dict__)
     model, _, _, _ = make_multimodal_multitask_model(**args.__dict__)
 
-    # good_structures has to be sorted by channel idx
-    good_channels = sorted([tm_out.channel_map[k] for k in args.structures_to_analyze])
-    good_structures = [[k for k in tm_out.channel_map.keys() if tm_out.channel_map[k] == v][0] for v in good_channels]
+    # the user can use '+' to create a new channel that merges other channels before postprocessing
+    # the user can use '++' to create a new channel that merges other channels after postprocessing
+    # those channels must be included alone in structures_to_analyze as well, then '+', then '++'
+    merged_locs = [
+        k for k in range(len(args.structures_to_analyze))
+        if '+' in args.structures_to_analyze[k] and '++' not in args.structures_to_analyze[k]
+    ]
+    merged_after_locs = [
+        k for k in range(len(args.structures_to_analyze))
+        if '++' in args.structures_to_analyze[k]
+    ]
+    uni_locs = [k for k in range(len(args.structures_to_analyze)) if '+' not in args.structures_to_analyze[k]]
+    assert((len(merged_locs) == 0) or (merged_locs[0] > uni_locs[-1]))
+    assert((len(merged_after_locs) == 0) or (merged_after_locs[0] > uni_locs[-1]))
+    merged_structures = [args.structures_to_analyze[k] for k in merged_locs]
+    merged_after_structures = [args.structures_to_analyze[k] for k in merged_after_locs]
+    uni_structures = [args.structures_to_analyze[k] for k in uni_locs]
+    merged_channels = [k.split('+') for k in merged_structures]
+    merged_after_channels = [k.split('++') for k in merged_after_structures]
+    for i in range(len(merged_channels)):
+        for j in range(len(merged_channels[i])):
+            merged_channels[i][j] = tm_out.channel_map[merged_channels[i][j]]
+    for i in range(len(merged_after_channels)):
+        for j in range(len(merged_after_channels[i])):
+            merged_after_channels[i][j] = tm_out.channel_map[merged_after_channels[i][j]]
+
+    # structures have to be in the same order as the channel map
+    good_channels = [tm_out.channel_map[k] for k in uni_structures]
+    assert (good_channels == sorted(good_channels))
```

Comment:
Lots of asserts which is fine, but I like the pattern `assert(condition, message)` so that if they fail the user gets a message and readers get a sense of why the assert is there.  No need to add messages to all of these, just a good pattern to strive for.